### ISSUE TITLE
Add storeEnhancer typings

### ISSUE THREAD

**Issue by BerkeleyTrue** [redux-observable/redux-observable#316]
We have a PR (#259) with working and tested code lacking documentation and typings.

We are looking for help creating typescript typings on this pr that replaces the epicMiddleware typings (should be very similar).

An isolated PR against master should do

**Comment by aminpaks** [comment]
I would be glad if can help writing all the typings for Typescript.
Haven’t yet read the new PR but will do and get back.

**Comment by aminpaks** [comment]
Can you tell me what's the remote branch to work on? I don't see the `BerkeleyTrue:feat/epic-enhancer` on your repo.

**Comment by jayphelps** [comment]
@aminpaks I would hold off for now. Right now I believe we no longer need to change from a middleware to an enhancer.

**Comment by aminpaks** [comment]
alright. But i'll be truly glad to contribute.

**Comment by BerkeleyTrue** [comment]
@jayphelps what's changed?

**Comment by jayphelps** [comment]
@BerkeleyTrue my recollection was that an enhancer was thought to be needed to provide state$ as a stream and to let epics synchronously emit an action on start-up, but I have working code for both of these requirements. Let me know if I missed something!

**Comment by BerkeleyTrue** [comment]
@jayphelps I see. Should I close these issues and the PR?

**Comment by BerkeleyTrue** [comment]
@jayphelps ping on this. Is there a PR for sync emission of actions?

**Comment by ShadowManu** [comment]
This issue originates from PR #259, targeted for issue #256, trying to solve issues that are not related to the state (or state$), but the order (and sometimes lack) of actions in `action$`.

@jayphelps ping on this :^) . AFAIK is not a problem solved by #410.

I'm very familiar with typescript typings, but not with the specific typings in here (store enhancers and the like). I'll take a look of it on the following days after #315 to help on this if this still applies.

**Comment by BerkeleyTrue** [comment]
@ShadowManu I'd hold back since a lot has changed in the rx and RO world that would require me to update my PR. I've held back on doing that until I've heard back from @jayphelps on whether the PR would be merged or not.

**Comment by ShadowManu** [comment]
Got it. I'll be on the lookout.
